src/schematools/types.py

```diff
@@ -1018,21 +1018,23 @@ def get_sub_fields(self, add_prefixes=False) -> Iterator[DatasetFieldSchema]:
         those subfields need to be prefixed with the name of the relation field.
         However, this is not the case for the so-called `dimension` fields
         of a temporal relation (e.g. `beginGeldigheid` and `eindGeldigheid`).
+
+        If self is not an object or array, the return value is an empty iterator.
         """
         from schematools.utils import toCamelCase
 
         field_name_prefix = ""
 
         if self.is_object:
-            # Field has direct sub fields (type=object)
+            # Field has direct subfields (type=object)
             required = set(self.get("required", []))
             properties = self["properties"]
         elif self.is_array_of_objects and self.field_items is not None:
             # Field has an array of objects (type=array, items are objects)
             required = set(self.field_items.get("required") or ())
             properties = self.field_items["properties"]
         else:
-            raise ValueError("Subfields are only possible for 'object' or 'array' fields.")
+            return ()
```

Comment:
This returns an `Iterable` not an `Iterator`. You should adjust the return type of this method.

Reply:
Hm. I am a bit hesistant about this one. When a field is not a relation- or nested field, the caller is not supposed to call `get_sub_fields` on that field, hence the `ValueError`. Returning an empty tuple silences this error. Maybe that is not a problem, but the caller is not informed that she is doing something weird. Could this not become problematic?

Reply:
With this behavior, both call sites in dso-api work without any checks or try-catch constructs. Any behavior we choose can cause problems, but empirically, this is safer than the current behavior (N=1, but still :)

Reply:
@gkoller Changed to `Iterable`.

Reply:
Ok, fine with me. Maybe good to  add a warning in the log.